src/status_im/ui/screens/routing/core.cljs

```diff
@@ -118,6 +118,13 @@
       (utils/update-if-present :initialRouteName name)
       (utils/update-if-present :mode name)))
 
+(defn new-get-state-for-action [default-get-state-for-action]
+  (fn [action state]
+    (let [event (get back-actions/back-actions @view-id)]
+      (when (and (= (.-type action) (.-BACK navigation/navigation-actions)) event (not= :default event))
+        (re-frame/dispatch [event]))
+      (default-get-state-for-action action state))))
```

Comment:
For now we're always popping the item from stack here, disregarding whatever might happen in `event` handler, but when Cancel processing is in place (and thus a possibility to block navigation action), how should this work? Currently `screens.navigation/processing-back-event?` exists, will there be some alternative?

Reply:
we could try to find an alternative after v1, for now Cancel screens just ignore hardware button or swipe